### PR TITLE
Added code format check to pre submit hook.

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:
This is a follow up PR for https://github.com/istio/pilot/pull/1078#issuecomment-323914300 @ldemailly 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by kyessenov** [comment]
Does this fail a PR if the code is not well formatted? Is this different from fmt linter?

**Comment by rshriram** [comment]
@gyliu513 we already have a linter in prow.

**Comment by ldemailly** [comment]
this wasn't covered by the linter, should be probably a commit hook to go fmt all go files
(@gyliu513 found files that weren't go fmted as part of https://github.com/istio/pilot/pull/1078/files )

**Comment by gyliu513** [comment]
Yes, the lint only check the syntax etc but not the format, so we still need this. @ldemailly @rshriram  @kyessenov any more comments for this?

File: prow/pilot-presubmit.sh [hunk 1]
```diff
@@ -50,6 +50,9 @@ bazel build //...
 echo '=== Go Build ==='
 ./bin/init.sh
 
+echo '=== Code Format Check ==='
+./bin/fmt.sh
```
Comment (ldemailly) [inline prow/pilot-presubmit.sh]: I left comments in the original PR that created this. but how is this going to do anything/'fail' if needed ?

Comment (kyessenov) [inline prow/pilot-presubmit.sh]: Indeed. This must fail if the code is not formatted.

Comment (gyliu513) [inline prow/pilot-presubmit.sh]: @ldemailly @kyessenov can you please show me for how to use Prow to trigger the presubmit script?

**Comment by rshriram** [comment]
temporarily closing to reduce load on test bot. sorry for the trouble @gyliu513 .. Please reopen in the morning.

**Comment by ldemailly** [comment]
somehow I think someone already got to it - I saw fmt errors on a PR last week

**Pull request closed without merging**